ceph_deploy/hosts/rhel/install.py

```diff
@@ -0,0 +1,94 @@
+from ceph_deploy.util import pkg_managers, templates
+from ceph_deploy.lib import remoto
+from ceph_deploy.hosts.util import install_yum_priorities
+
+
+def rpm_dist(distro):
+    return 'el' + distro.normalized_release.major
+
+
+def repository_url_part(distro):
+    return 'rhel' + distro.normalized_release.major
+
+
+def install(distro, version_kind, version, adjust_repos):
+    logger = distro.conn.logger
+    release = distro.release
+    machine = distro.machine_type
+    repo_part = repository_url_part(distro)
+    dist = rpm_dist(distro)
+
+    pkg_managers.yum_clean(distro.conn)
+    pkg_managers.yum(distro.conn, ['ceph', 'ceph-mon', 'ceph-osd'])
+
+
+def mirror_install(distro, repo_url, gpg_url, adjust_repos, extra_installs=True):
+    repo_url = repo_url.strip('/')  # Remove trailing slashes
+    gpg_url_path = gpg_url.split('file://')[-1]  # Remove file if present
+
+    pkg_managers.yum_clean(distro.conn)
+
+    if adjust_repos:
+        remoto.process.run(
+            distro.conn,
+            [
+                'rpm',
+                '--import',
+                gpg_url_path,
+            ]
+        )
+
+        ceph_repo_content = templates.ceph_repo.format(
+            repo_url=repo_url,
+            gpg_url=gpg_url
+        )
+
+        distro.conn.remote_module.write_yum_repo(ceph_repo_content)
+
+    if extra_installs:
+        pkg_managers.yum(distro.conn, ['ceph', 'ceph-mon', 'ceph-osd'])
+
+
+def repo_install(distro, reponame, baseurl, gpgkey, **kw):
+    logger = distro.conn.logger
+    # Get some defaults
+    name = kw.pop('name', '%s repo' % reponame)
```

Comment:
Hmm, this part was a straight lift from hosts/centos/install.py.

I believe that is something worth looking at.  The mutated `kw` is passed to templates.custom_repo() a bit further down, so it might have some importance.  I'll make a not of it to check out later.